addons/binding/org.openhab.binding.sleepiq/src/main/java/org/openhab/binding/sleepiq/handler/SleepIQCloudHandler.java

```diff
@@ -0,0 +1,322 @@
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.sleepiq.handler;
+
+import static org.openhab.binding.sleepiq.SleepIQBindingConstants.THING_TYPE_CLOUD;
+import static org.openhab.binding.sleepiq.config.SleepIQCloudConfiguration.USERNAME;
+import static org.openhab.binding.sleepiq.config.SleepIQCloudConfiguration.PASSWORD;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.config.core.status.ConfigStatusMessage;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.ConfigStatusBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.sleepiq.SleepIQBindingConstants;
+import org.openhab.binding.sleepiq.config.SleepIQCloudConfiguration;
+import org.openhab.binding.sleepiq.internal.SleepIQConfigStatusMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.syphr.sleepiq.api.Configuration;
+import org.syphr.sleepiq.api.LoginException;
+import org.syphr.sleepiq.api.SleepIQ;
+import org.syphr.sleepiq.api.UnauthorizedException;
+import org.syphr.sleepiq.api.model.Bed;
+import org.syphr.sleepiq.api.model.BedStatus;
+import org.syphr.sleepiq.api.model.FamilyStatus;
+
+import com.google.common.base.Objects;
+
+/**
+ * The {@link SleepIQCloudHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Gregory Moyer - Initial contribution
+ */
+public class SleepIQCloudHandler extends ConfigStatusBridgeHandler {
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPE_UIDS = Collections.singleton(THING_TYPE_CLOUD);
+
+    private final Logger logger = LoggerFactory.getLogger(SleepIQCloudHandler.class);
+
+    private List<BedStatusListener> bedStatusListeners = new CopyOnWriteArrayList<>();
+
+    private ScheduledFuture<?> pollingJob;
+
+    private Runnable pollingRunnable = new Runnable() {
+
+        @Override
+        public void run() {
+            publishBedStatusUpdates();
+        }
+    };
+
+    private SleepIQ cloud;
+
+    public SleepIQCloudHandler(final Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void initialize() {
+
+        try {
+            createCloudConnection();
```

Comment:
If you put line 99 - 101 in here, you don't need the `return`'s and you error handling is more separated from your code flow.